vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.jsp

```diff
@@ -52,43 +67,144 @@ $(document).ready(function() {
 		$("#addTextAlert").hide();	
 	});
 	
+	$("#cancelDeleteText").click(function() {
+		$("#confirmDeleteTextAlert").hide();
+	});
+	
+	$("#cancelDeleteImage").click(function() {
+		$("#confirmDeleteImageAlert").hide();
+	});
+	
+	$("#cancelDelete").click(function() {
+		$("#confirmDeleteImageAlert").hide();	
+	});
+	
 	$("#cancelImageBtn").click(function() {
 		$("#image1").remove();
 		$("#addImgAlert").hide();	
 	});
 	
+	
 	$("#submitText").on("click", function(e) {
 		$("#addTextAlert").hide();
 		e.preventDefault();
 
 		var payload = {};
 		payload["content"] = $("#textBlockText").val();
-		
-		var textblock = $('<div class="card card-body">'+payload["content"]+'</div>');
-		$(textblock).css({
-			'margin': "10px"
-		});
-		$('#slideSpace').append(textblock);
 		++contentCount;
 		payload["contentOrder"] = contentCount;
 		// ------------- creating text content blocks ------------
-		$.post("<c:url value="/staff/module/${module.id}/slide/${slide.id}/textcontent?${_csrf.parameterName}=${_csrf.token}" />", payload, function(data) {
+		$.ajax({
+		    url: "<c:url value="/staff/module/${module.id}/slide/${slide.id}/textcontent?${_csrf.parameterName}=${_csrf.token}" />",
+		    type: 'POST',
+		    data: payload,
+		    success: function(data) {
+		    	var textBlock = $('<div class="valueDiv card card-body"><div class="row"><div class="col"><p>'+payload["content"]+'</p></div><div class="col"><input type="hidden" id="deleteTextId"><input class="btn btn-danger deleteText" type="submit" value="Delete" style="float: right;"></div></div></div>');
+				$(textBlock).css({
+					'margin': "10px"
+				});
+				$('#slideSpace').append(textBlock);
+		    	$(textBlock).attr( 'id', data["textBlock"]);
+		        $('#'+data["textBlock"]+ ' #deleteTextId').attr( 'value', data["textBlock"]);
+		        
+		    },
+			error: function(data) {
+				--contentCount;
+				var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try to submit again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+				$('.error').append(alert); 
+				$(".error").delay(4000).slideUp(500, function(){
+				    $(".error").empty();
+				});
+				$("#textBlockText").val('');
+			}
 		});
+
+		
+		$("#textBlockText").val('');
+	});
+	
+	// ---------- Delete Text Block -------------
+	
+		$("#deleteText").on("click", function(e) {
+		e.preventDefault();
+		$("#confirmDeleteTextAlert").hide();
+		var payload = {};
+		payload["blockId"] = $('#deleteTextId').attr('value');
+		$('#deleteTextId').remove()
 		
-		$("#textBlockText").val('')
+		// ------------- delete text content blocks ------------
+		$.ajax({
+		    url: "<c:url value="/staff/module/${module.id}/slide/${slide.id}/textcontent/" />" + payload["blockId"] + "?${_csrf.parameterName}=${_csrf.token}",
+		    type: 'DELETE',
+		    data: payload,
```

Comment:
since payload has only the block id and the block id is submitted via the URL, do you need this here?